### ISSUE TITLE
Social login - redirect loop

### ISSUE THREAD

**Issue by mferak** [CakeDC/users#329]
When logging in via the social media buttons, the page gets into a redirect loop.

This only happens if the log in is initialised from the '/login' page. 
Otherwise there's no redirect loop and everything works fine.

**Comment by mferak** [comment]
Appears to be the same issue as https://github.com/CakeDC/users/issues/316

**Comment by mferak** [comment]
I found out that while regular login uses the Auth.loginRedirect URL after successful login, the social login auth redirects to referer(). Going to override the LoginTrait login function to see if that could solve it for the time being.

**Comment by steinkel** [comment]
feel free to submit a PR if you fix it before I can reproduce and fix it :+1:

**Comment by vietkungfu** [comment]
In my case:
+ bootstrap.php: have to write configure before load plugin

>`Configure::write('Users.config', ['users']);`
>`Plugin::load('CakeDC/Users', ['routes' => true, 'bootstrap' => true]);`

+ permissions.php:

>```
<?php
    return [
      'Users.SimpleRbac.permissions' => [
          [
            'role' => '*',
            'plugin' => 'CakeDC/Users',
            'controller' => 'Users',
            'action' => '*'
        ]
      ]
   ];
?>
```

+ routes.php: 

>`Router::connect('/login/*', array('plugin' => 'CakeDC/Users', 'controller' => 'Users', 'action' => 'login'));`

**Comment by mferak** [comment]
Connected with https://github.com/CakeDC/users/pull/310 ?

**Comment by vietkungfu** [comment]
@mferak : Just in my case.

**Comment by steinkel** [comment]
closed per PR #342
